File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/ObeziteIzlemFormViewModel.ts
//$6C1D9370
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { ObeziteIzlemVeriSeti } from 'NebulaClient/Model/AtlasClientModel';
import { ObeziteEkHastalik } from 'NebulaClient/Model/AtlasClientModel';
import { ObeziteEgzersiz } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSPsikolojikTedavi } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSObeziteIlacTedavisi } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSMorbidObezHastaLenfatikOdemVarligi } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSDiyetTedavisiTibbiBeslenmeTedavisi } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSICD } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSEgzersiz } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';

export class ObeziteIzlemFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.ObeziteIzlemFormViewModel, Core';
     @Type(() => ObeziteIzlemVeriSeti)
    public _ObeziteIzlemVeriSeti: ObeziteIzlemVeriSeti = new ObeziteIzlemVeriSeti();
     @Type(() => ObeziteEkHastalik)
    public ObeziteEkHastalikGridList: Array<ObeziteEkHastalik> = new Array<ObeziteEkHastalik>();
     @Type(() => ObeziteEgzersiz)
    public ObeziteEgzersizGridList: Array<ObeziteEgzersiz> = new Array<ObeziteEgzersiz>();
     @Type(() => SKRSPsikolojikTedavi)
    public SKRSPsikolojikTedavis: Array<SKRSPsikolojikTedavi> = new Array<SKRSPsikolojikTedavi>();
     @Type(() => SKRSObeziteIlacTedavisi)
     public SKRSObeziteIlacTedavisis: Array<SKRSObeziteIlacTedavisi> = new Array<SKRSObeziteIlacTedavisi>();
     @Type(() => SKRSMorbidObezHastaLenfatikOdemVarligi)
     public SKRSMorbidObezHastaLenfatikOdemVarligis: Array<SKRSMorbidObezHastaLenfatikOdemVarligi> = new Array<SKRSMorbidObezHastaLenfatikOdemVarligi>();
     @Type(() => SKRSDiyetTedavisiTibbiBeslenmeTedavisi)
     public SKRSDiyetTedavisiTibbiBeslenmeTedavisis: Array<SKRSDiyetTedavisiTibbiBeslenmeTedavisi> = new Array<SKRSDiyetTedavisiTibbiBeslenmeTedavisi>();
     @Type(() => SKRSICD)
     public SKRSICDs: Array<SKRSICD> = new Array<SKRSICD>();
     @Type(() => SKRSEgzersiz)
    public SKRSEgzersizs: Array<SKRSEgzersiz> = new Array<SKRSEgzersiz>();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/E_Nabiz_Modulu/ObeziteIzlemFormViewModel.cs
//$82AF8012
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Infrastructure.Helpers;

namespace Core.Controllers
{
    public partial class ObeziteIzlemVeriSetiServiceController
    {
        partial void PreScript_ObeziteIzlemForm(ObeziteIzlemFormViewModel viewModel, ObeziteIzlemVeriSeti obeziteIzlemVeriSeti, TTObjectContext objectContext)
        {
            Guid? selectedEpisodeActionID = viewModel.GetSelectedEpisodeActionID();
            if (selectedEpisodeActionID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(selectedEpisodeActionID.Value);
                viewModel._ObeziteIzlemVeriSeti.EpisodeAction = episodeAction;
                viewModel._ObeziteIzlemVeriSeti.SubEpisode = episodeAction.SubEpisode;

            }
            ContextToViewModel(viewModel, objectContext);
        }

    }
}

namespace Core.Models
{
    public partial class ObeziteIzlemFormViewModel : BaseViewModel, IENabizViewModel
    {
        public void AddENabizObjectViewModelToContext(TTObjectContext objectContext)
        {
            objectContext.AddObject(this._ObeziteIzlemVeriSeti);

            if (this.ObeziteEgzersizGridList != null)
            {
                foreach (var item in this.ObeziteEgzersizGridList)
                {
                    var egzersizImported = (ObeziteEgzersiz)objectContext.AddObject(item);
                    egzersizImported.ObeziteIzlemVeriSeti = this._ObeziteIzlemVeriSeti;
                }
            }

            if (this.ObeziteEkHastalikGridList != null)
            {
                foreach (var item in this.ObeziteEkHastalikGridList)
                {
                    var ekHastalikImported = (ObeziteEkHastalik)objectContext.AddObject(item);
                    ekHastalikImported.ObeziteIzlemVeriSeti = this._ObeziteIzlemVeriSeti;
                }
            }

            var obeziteIzlem = (ObeziteIzlemVeriSeti)objectContext.GetLoadedObject(this._ObeziteIzlemVeriSeti.ObjectID);


            new SendToENabiz(objectContext, obeziteIzlem.SubEpisode, obeziteIzlem.ObjectID, obeziteIzlem.ObjectDef.Name, "240", Common.RecTime());
        }
    }
}
